### ISSUE TITLE
Improve accessibility of the "Save and publish" dialog

### ISSUE THREAD

**Issue by BatJan** [umbraco/Umbraco-CMS#5882]
A brief description of the issue goes here.

This issue is related to issue no. 5 in the list of accessibility issues here https://github.com/umbraco/Umbraco-CMS/issues/5277

While it appears that the dropdown is now keyboard accessible there still are a few other concerns that needs to be addressed in order to improve screen reader accessibility.

1. For semantics the `<a>` tags should be changed to `<button>` tags
2. The aria-expanded and aria-controls attributes should be added to the caret communicating the state of the "popup" to screen readers
3. The aria-hidden attribute should be added to the button group communicating if the options are visible or not to screen readers
4. A visually hidden text of "More publishing options" or similar should be added to the "caret" element providing screen readers some proper context about what the button does
5. The ellipsis should be wrapped in a `<span>` element with the aria-hidden attribute set to true since the ellipsis does not make sense in a non-visual context. Maybe it does not in a visual context either? But that's not to be addressed in this issue

I'll create a draft PR for this issue.
